src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
```

Comment:
PullProofAI

Consider adding input validation to ensure `context` is not null or undefined before using it.
```typescript
if (!context) {
    throw new Error('Invalid execution context');
}
```


<!-- This is an auto-generated comment by PullProofAI -->